tests/integration/istioctl/analyze_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioctl
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+const (
+	serviceRoleBindingFile = "testdata/servicerolebinding.yaml"
+	serviceRoleFile        = "testdata/servicerole.yaml"
+)
+
+func TestEmptyCluster(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			g := NewGomegaWithT(t)
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-analyze",
+				Inject: true,
+			})
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			// For a clean istio install with injection enabled, expect no validation errors
+			output := runIstioctl(t, istioCtl,
+				[]string{"experimental", "analyze", "--namespace", ns.Name(), "--use-kube"})
```

Comment:
It would be useful to add this capability to the common istioctl component. Maybe replace the existing `Invoke` method in the `Instance` interface with something that takes format+args? e.g.

```golang
istioct.NewOrFail(t, ctx, istioctl.Config{})

out, err := istioctl.Exec("x analyze --namespace %v --use-kube", ns.Name)
if err != nil {
    // etc ..
}

// or

output := istioctl.ExecOrDie(t, "x analyze --namespace %v --use-kube", ns.Name)
g.Expect(output).To(BeEmpty())
```

Reply:
Created InvokeOrFail, following the usual pattern. 

Kept the local helper for the line split functionality, which is more specific to this test.